google-services/Location/LocationAddress/LocationAddress/MainActivity.cs

```diff
@@ -1,186 +1,292 @@
+        }
+
+        public bool CheckPermissions()
+        {
+            var permissionState = ContextCompat.CheckSelfPermission(this, Manifest.Permission.AccessFineLocation);
+            return permissionState == (int)Permission.Granted;
+        }
+
+        public void RequestPermissions()
+        {
+            bool shouldProvideRationale = ActivityCompat.ShouldShowRequestPermissionRationale(this, Manifest.Permission.AccessFineLocation);
+
+            if (shouldProvideRationale)
+            {
+                Log.Info(TAG, "Displaying permission rationale to provide additional context.");
+                ShowSnackbar(Resource.String.permission_rationale, Android.Resource.String.Ok, new AddressResultReceiver.RequestPermissionsOnClickListener { Activity = this });
+            }
+            else
+            {
+                Log.Info(TAG, "Requesting permission");
+                ActivityCompat.RequestPermissions(this, new[] { Manifest.Permission.AccessFineLocation }, RequestPermissionsRequestCode);
+            }
+        }
+    }
+
+    public class GetAddressOnSuccessListener : Object, IOnSuccessListener
+    {
+
+        public LocationAddress.MainActivity Activity { get; set; }
+
+        public void OnSuccess(Object location)
+        {
+            if (location == null)
+            {
+                Log.Warn(Activity.TAG, "onSuccess:null");
+                return;
+            }
+
+            Activity.mLastLocation = (Location)location;
+
+            if (!Geocoder.IsPresent)
+            {
+                Activity.ShowSnackbar(Activity.GetString(Resource.String.no_geocoder_available));
+                return;
+            }
+
+            if (Activity.mAddressRequested)
+            {
+                Activity.StartIntentService();
+            }
+        }
+    }
+
+    public class GetAddressOnFailureListener : Java.Lang.Object, IOnFailureListener
+    {
+
+        public LocationAddress.MainActivity Activity { get; set; }
+
+
+        public void OnFailure(Exception e)
+        {
+            Log.Warn(Activity.TAG, "getLastLocation:onFailure", e);
+        }
+    }
+
+    public class AddressResultReceiver : ResultReceiver
+    {
+        public MainActivity Activity { get; set; }
+        public AddressResultReceiver(Handler handler) : base(handler) { }
+
+        protected override void OnReceiveResult(int resultCode, Bundle resultData)
+        {
+            Activity.mAddressOutput = resultData.GetString(Constants.ResultDataKey);
+            Activity.DisplayAddressOutput();
+
+            if (resultCode == Constants.SuccessResult)
+            {
+                Activity.ShowToast(Activity.GetString(Resource.String.address_found));
+            }
+
+            Activity.mAddressRequested = false;
+            Activity.UpdateUiWidgets();
+        }
+
+        public void OnRequestPermissionsResult(int requestCode, string[] permissions, int[] grantResults)
```

Comment:
I believe this needs to be an override, which means it will need to move up to be a member of the MainActivity class.

Reply:
Moved in MainActivity class. Modified with override keyword.